luminosityleds/server
# LL-317

# Description

In this new PR/Ticket, routes and controllers have been added to the routes and controllers folders via various files for the ActiveMQ notification service. This enables different URL's to lead to different results for the service such as viewing user accounts, creating them, deleting them, etc.

Subsequent # (issue): LL-317

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

Steps (For Local Host):

1. Run Docker Desktop
2. Make sure you are in the root directory (server)
3. Run "npm i" if needed (root directory and possibly publish/subscribe directories as well)
4. Run "docker compose up --build" (You will need the correct .env file.  Ask @KevinLemon112 or @pogi7 for it)
5. Open Postman downloaded app (browser Postman does NOT work for Local Host I believe...)
6. Input the following routes to test (check MongoDB to confirm changes): 
 
# User Routes

**Register User**

URL: http://localhost:4000/users/register
Method: POST
Body: raw - JSON

JSON (copy and paste this into Postman in the raw - JSON section):

{
"email": "example@example.com",
"name": "John Doe",
"creationDate": "2023-08-28T05:33:57.702Z",
"deletionDate": "2023-08-29T05:33:57.702Z", // Optional, can be omitted if not needed
"lastUpdated": "2023-08-30T05:33:57.702Z",
"devicesLinked": []
}

OR (if you want to test linking a device)

{
  "email": "example@example.com",
  "name": "John Doe",
  "creationDate": "2023-08-28T05:33:57.702Z",
  "deletionDate": "2023-08-29T05:33:57.702Z", // Optional, can be omitted if not needed
  "lastUpdated": "2023-08-30T05:33:57.702Z",
  "devicesLinked": [
    {
      "uuid": "device1_uuid",
      "lastUpdated": "2023-08-28T05:33:57.702Z",
      "powered": true, // Optional, can be omitted if not needed
      "poweredTimestamp": "2023-08-28T05:33:57.702Z", // Optional, can be omitted if not needed
      "connected": true, // Optional, can be omitted if not needed
      "connectedTimestamp": "2023-08-28T05:33:57.702Z", // Optional, can be omitted if not needed
      "color": "#FFFFFF", // Optional, can be omitted if not needed
      "colorTimestamp": "2023-08-28T05:33:57.702Z", // Optional, can be omitted if not needed
      "brightness": 50, // Optional, can be omitted if not needed
      "brightnessTimestamp": "2023-08-28T05:33:57.702Z" // Optional, can be omitted if not needed
    }
  ]
}

-----------------------------------------------------------------------------------------------------------------------------------------

**Get All Users**

URL: http://localhost:4000/users/all
Method: GET
Get User by ID

-----------------------------------------------------------------------------------------------------------------------------------------

**Get User by ID**

URL: http://localhost:4000/users/:id
Method: GET

(replace ":id" with actual MongoDB account ID)

-----------------------------------------------------------------------------------------------------------------------------------------

**Delete User by ID**

URL: http://localhost:4000/users/:id/delete
Method: DELETE

(replace ":id" with actual MongoDB account ID)

-----------------------------------------------------------------------------------------------------------------------------------------

# Publish/Subscribe Routes

**Publish Message**

URL: http://localhost:6000/publish/:id
Method: GET

(replace ":id" with any ID)


-----------------------------------------------------------------------------------------------------------------------------------------

**Subscribe Status**

URL: http://localhost:7000/subscribe/status
Method: GET

-----------------------------------------------------------------------------------------------------------------------------------------

# Checklist (when relevant):


[TypeScript](https://google.github.io/styleguide/tsguide.html)
[Python](https://peps.python.org/pep-0008/)
[Schema](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/forefront-2010/ee652242(v=vs.100))
- [x] I have included a reviewer on the pull request [*Replace this text with the reviewer*]
